### PR TITLE
Bug Fix: Couldn't save income determination after form error on create record

### PR DESCRIPTION
**Pull request by nickbristow**
## Background
In the add record flow, if you failed to put all information on the income determination section, correcting the form would not allow you to save the form. 

## GitHub Issue
#1394 

## Validation Plan
Please outline the steps you've already taken to validate this body of work, along with any additional steps that need to be taken in order to complete testing.

## Automated Testing
- [x] All unit tests are passing.

### CODE REVIEW

File: client/src/components/Forms/FamilyIncome/Form.tsx [hunk 1]
```diff
@@ -128,12 +123,22 @@ export const FamilyIncomeForm: React.FC<FamilyIncomeFormProps> = ({
...
+
+    const saveData =
+      determination.id && !redetermine
+        ? updateDetermination
+        : createDetermination;
+
     saveData(updatedData)
       .then(afterSaveSuccess)
       .catch((err) => {
-        console.error(err);
```
Comment (vonbearshark) [inline client/src/components/Forms/FamilyIncome/Form.tsx:134]: Do we want to rm these? Genuinely don't know. I do see it used elsewhere, and keeping console _error_ logging seems fine to me

**Review by vonbearshark — APPROVED** [review]

File: client/src/components/Forms/FamilyIncome/Form.tsx (hunk 1 above)
Comment (JordanGuinn) [reply client/src/components/Forms/FamilyIncome/Form.tsx:134]: Eh, I don't mind keeping them around personally

File: client/src/components/Forms/FamilyIncome/Form.tsx [hunk 2]
```diff
@@ -128,12 +123,22 @@ export const FamilyIncomeForm: React.FC<FamilyIncomeFormProps> = ({
 
   const onSubmit = (updatedData: IncomeDetermination) => {
     setLoading(true);
+    if (!determination.id && inCreateFlow && dets[0])
```
Comment (JordanGuinn) [inline client/src/components/Forms/FamilyIncome/Form.tsx:126]: What exactly is the fix here?

File: client/src/components/Forms/FamilyIncome/Form.tsx (hunk 1 above)
Comment (nickbristow) [reply client/src/components/Forms/FamilyIncome/Form.tsx:134]: I thought i added that in my debugging, adding back in

File: client/src/components/Forms/FamilyIncome/Form.tsx (hunk 2 above)
Comment (nickbristow) [reply client/src/components/Forms/FamilyIncome/Form.tsx:126]: if we don't have an id on the determination object and we are in the create flow and we have a determination on the child, it means the child has already had one created, so use that id.

**Review by JordanGuinn — APPROVED** [review]

**Comment by JordanGuinn** [comment]
## Validation Plan
- [ ] Confirm I can **add** a new income determination to a child from the edit view
- [ ] Confirm I can **edit** an existing determination for a child from the edit view
- [ ] Confirm I can add a new income determination to a child from the create view, under the following conditions
   - [ ] When all fields are present
   - [ ] When certain fields are missing

**Comment by JordanGuinn** [comment]
All looks good to me - this is ready

**Pull request merged**